### ISSUE TITLE
kvserver: consult quota pool before lease transfer

### ISSUE THREAD

**Issue by tbg** [cockroachdb/cockroach#95992]
**Is your feature request related to a problem? Please describe.**

See https://github.com/cockroachlabs/support/issues/2008. We think the main culprit in that issue were lease transfers to nodes that ended up needing a snapshot, which should[^1] be fixed in `22.2` at least in the common case (and further de-risked by transferring via expiration-based leases), but it's still a bad idea to send the lease to a node that is not actively being enforced by the quota pool. If a peer is enforced by the quota pool, we shouldn't see it get cut off from the log via a log truncation either, which provides additional protection against theoretical races where the snapshot might become necessary "just after" we initiate the lease transfer.

It's possible that we'll never again see issues after #82758 is picked up by all customers.

[^1]: https://github.com/cockroachdb/cockroach/pull/82758

Jira issue: CRDB-23885

**Comment by tbg** [comment]
This issue is obsolete if we disable/remove the quota pool, x-ref https://github.com/cockroachdb/cockroach/issues/106063

**Comment by nvanbenschoten** [comment]
Closing as duplicate of https://github.com/cockroachdb/cockroach/issues/96304.
